tiled/adapters/tiff.py

```diff
@@ -135,160 +134,6 @@ def structure(self) -> ArrayStructure:
-
-    def metadata(self) -> JSON:
-        """
-
-        Returns
-        -------
-
-        """
-        # TODO How to deal with the many headers?
-        return self._provided_metadata
-
-    def read(self, slice: Optional[NDSlice] = ...) -> NDArray[Any]:
-        """Return a numpy array
-
-        Receives a sequence of values to select from a collection of tiff files
-        that were saved in a folder The input order is defined as: files -->
-        vertical slice --> horizontal slice --> color slice --> ... read() can
-        receive one value or one slice to select all the data from one file or
-        a sequence of files; or it can receive a tuple (int or slice) to select
-        a more specific sequence of pixels of a group of images.
-
-        Parameters
-        ----------
-        slice :
-
-        Returns
-        -------
-                Return a numpy array
-
-        """
-
-        if slice is Ellipsis:
-            return self._seq.asarray()
-        if isinstance(slice, int):
-            # e.g. read(slice=0) -- return an entire image
-            return tifffile.TiffFile(self._seq.files[slice]).asarray()
-        if isinstance(slice, builtins.slice):
-            # e.g. read(slice=(...)) -- return a slice along the image axis
-            return tifffile.TiffSequence(self._seq.files[slice]).asarray()
-        if isinstance(slice, tuple):
-            if len(slice) == 0:
-                return self._seq.asarray()
-            if len(slice) == 1:
-                return self.read(slice=slice[0])
-            image_axis, *the_rest = slice
-            # Could be int or slice (0, slice(...)) or (0,....); the_rest is converted to a list
-            if isinstance(image_axis, int):
-                # e.g. read(slice=(0, ....))
-                arr = tifffile.TiffFile(self._seq.files[image_axis]).asarray()
-            elif image_axis is Ellipsis:
-                # Return all images
-                arr = tifffile.TiffSequence(self._seq.files).asarray()
-                the_rest.insert(0, Ellipsis)  # Include any leading dimensions
-            elif isinstance(image_axis, builtins.slice):
-                arr = self.read(slice=image_axis)
-            arr = np.atleast_1d(arr[tuple(the_rest)])
-            return arr
-
-    def read_block(
-        self, block: Tuple[int, ...], slice: Optional[NDSlice] = ...
-    ) -> NDArray[Any]:
-        """
-
-        Parameters
-        ----------
-        block :
-        slice :
-
-        Returns
-        -------
-
-        """
-        if any(block[1:]):
-            raise IndexError(block)
-        arr = self.read(builtins.slice(block[0], block[0] + 1))
-        return arr[slice]
-
-    def structure(self) -> ArrayStructure:
-        """
-
-        Returns
-        -------
-
-        """
-        return self._structure
+class TiffSequenceAdapter(FileSequenceAdapter):
+    def _load_from_files(self, slc: Union[slice, int] = slice(None)) -> NDArray[Any]:
```

Comment:
It is convention to use `slice` as the parameter name. We use `builtins.slice` to avoid a clash. Arguable, but might as well be consistent.

Reply:
Fixed!